### PR TITLE
Basic Sorting Feature implemented

### LINKED ISSUE #27: As a student, I can sort the contacts by their name, phone, email or address
... so that I can organise my contacts list.

For example:
`sort n/` sorts contacts by their names in alphabetical order
`sort !p/` sorts contacts by their phone numbers in decreasing order (`!` reverses the default order)

### PR DESCRIPTION
**Pull request by Eugene-Ong-W-X**
`sort n/`, `sort p/`, `sort e/`, `sort a/` now works
`sort !n/`, `sort !p/`, `sort !e/`, `sort !a/` also now works (sorting in reverse order)

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            model.sortByAddress();
+        } else {
+            throw new CommandException(MESSAGE_WRONG_PREFIX);
+        }
+
+        if (isReverse) {
+            model.reverseSort();
+        }
+
+        return new CommandResult(MESSAGE_SUCCESS + parameter.getPrefix());
```
Comment (ChongCheeKaiClarence) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:67]: I notice there are some common traits among the conditions. Could a switch statement be used instead?

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 2]
```diff
@@ -97,6 +98,45 @@ public void setPersons(List<Person> persons) {
...
+    /**
+     * Sorts the list by address, alphabetically.
+     */
+    public void sortByAddress() {
+        SortedList<Person> sorted = internalList.sorted((curr, next) -> curr.getAddress().compareTo(next.getAddress()));
+        internalList.setAll(sorted);
+    }
+
+    /**
+     * Sorts the list in reverse order.
```
Comment (ChongCheeKaiClarence) [inline src/main/java/seedu/address/model/person/UniquePersonList.java:134]: I like your usage of lambda functions. Makes it short and simple.

File: src/main/java/seedu/address/logic/commands/SortCommand.java (hunk 1 above)
Comment (Eugene-Ong-W-X) [reply src/main/java/seedu/address/logic/commands/SortCommand.java:67]: Can't use switch here as switch only works with primitives, Strings, enums, etc

Comment (ChongCheeKaiClarence) [reply src/main/java/seedu/address/logic/commands/SortCommand.java:67]: Oh. Ok, I see no other issues. Thanks for clarifying.

**Review by ChongCheeKaiClarence — APPROVED** [review]
LGTM

**Pull request merged**